deployment/7_upgrade_ens.js

```diff
@@ -0,0 +1,31 @@
+const TestENSRegistry = require('../build/TestENSRegistry');
+const TestReverseRegistrar = require('../build/TestReverseRegistrar');
+const ENSResolver = require('../build/ENSResolver');
+const ENSManager = require('../build/ENSManager');
+
+const deploy = async (network) => {
+    const config = configurator.config;
+    const domain = config.ENS.domain;
+    const deploymentAccount = await deployer.signer.getAddress();
+
+	// ENS domain
+	const parts = domain.split('.');
+	const extension = parts[1];
+	const domainName = parts[0];
+
+    // Instantiate the ENS Registry
+    const ensRegistry = await deployer.wrapDeployedContract(TestENSRegistry, config.ENS.ensRegistry);
+
+	// Create the 'argentx.xyz' wallet ENS namespace
+	const setSubnodeOwnerArgent = await ensRegistry.contract.setSubnodeOwner(utils.namehash(extension), utils.sha3(domainName), deploymentAccount);
+	await ensRegistry.verboseWaitForTransaction(setSubnodeOwnerArgent, `Setting Subnode Owner for ${domainName}.${extension}`);
+
+    const ensResolver = await deployer.deploy(ENSResolver);
+    const ensManager = await deployer.deploy(ENSManager, {}, domain, walletNode, config.ENS.ensRegistry, ensResolver.contractAddress);
```

Comment:
Once the new ENSManager is deployed we need to add the WalletFactory as a manager so that the factory can register subdomains to the wallets it creates:
```
    const ENSManagerAddManagerTx = await ENSManagerWrapper.contract.addManager(config.contracts.WalletFactory);
    await ENSManagerWrapper.verboseWaitForTransaction(ENSManagerAddManagerTx, 'Set the WalletFactory as the manager of the ENS Manager');
```
We also need to change the address of the ENSMAnager in the factory:
```
        const WalletFactoryChangeENSManagerTx = await WalletFactoryWrapper.contract.changeENSManager(ENSManagerWrapper.contractAddress);
        await WalletFactoryWrapper.verboseWaitForTransaction(WalletFactoryChangeENSManagerTx, `Change the ENSManager in the WalletFactory`);
```
Then we need to et the Multisig as the owner of the ENSManager:
```
        const changeOwnerTx = await ENSManagerWrapper.contract.changeOwner(config.contracts.MultiSigWallet);
        await ENSManagerWrapper.verboseWaitForTransaction(changeOwnerTx, `Set the MultiSig as the owner of ENSManagerWrapper}`);
```

Reply:
The `WalletFactory` itself would have changed as well right?

Reply:
No. We anticipated that the ENSManager might need to be changed in the future so we can update  the address of the ENSManager in the factory without redeploying it.

Reply:
It's misleading because the `ArgentENSManager` is imported but only the interface definition `IENSManager` inside that file is actually used. I have split the interface into its own file to avoid future confusion.